src/raftstore/store/store.rs

```diff
@@ -1100,6 +1100,12 @@ impl<T: Transport, C: PdClient> Store<T, C> {
             return Err(box_err!("mismatch peer id {} != {}", peer.peer_id(), peer_id));
         }
 
+        let header = msg.get_header();
+        // If header's term is 2 verions behind current term, leadership may have been changed away.
+        if header.get_term() > 0 && peer.term() > header.get_term() + 1 {
```

Comment:
term may increase without leader changed, we have already checked it is leader, so this case is okay.